roles/openshift_cloud_provider/tasks/vsphere.yml

```diff
@@ -3,4 +3,26 @@
   template:
     dest: "{{ openshift.common.config_base }}/cloudprovider/vsphere.conf"
     src: vsphere.conf.j2
-  when: openshift_cloudprovider_vsphere_username is defined and openshift_cloudprovider_vsphere_password is defined and openshift_cloudprovider_vsphere_host is defined and openshift_cloudprovider_vsphere_datacenter is defined and openshift_cloudprovider_vsphere_datastore is defined
+  when: 
+  - openshift_cloudprovider_vsphere_username is defined 
+  - openshift_cloudprovider_vsphere_password is defined 
+  - openshift_cloudprovider_vsphere_host is defined 
+  - openshift_cloudprovider_vsphere_datacenter is defined 
+  - openshift_cloudprovider_vsphere_datastore is defined
+
+- name: configure vsphere svc account
+  include_role:
+    name: openshift_cloud_provider
+    tasks_from: vsphere-svc
+  when:
+  - openshift_cloudprovider_kind is defined
+  - openshift_cloudprovider_kind == 'vsphere'
+  - openshift_version | version_compare('3.9', '>=')
+  - inventory_hostname == openshift_master_hosts[0]
+
+- name: update vsphere provider master config
+  include_tasks: update-vsphere.yml
```

Comment:
update-vsphere.yml is in the openshift_control_plane role so need to import that role or consider if it should be moved from openshift_control_plane to openshift_cloud_provider. If so then we need to update playbooks/openshift-master/private/upgrade.yml that references this task too

```
  - name: update vsphere provider master config
    import_role:
      name: openshift_control_plane
      tasks_from: update-vsphere
```

Reply:
Good call. I'll move it and make sure that the upgrade calls the cloud provider as well,

Reply:
In the upgrade.yml playbook, I've changed the reference to cloud_provider.